packages/react/src/components/Toolbar/ToolbarItem.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import * as PropTypes from 'prop-types'
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+
+import {
+  UIComponent,
+  createShorthandFactory,
+  UIComponentProps,
+  ChildrenComponentProps,
+  ContentComponentProps,
+  commonPropTypes,
+  childrenExist,
+  isFromKeyboard,
+} from '../../lib'
+import { ComponentEventHandler, ShorthandValue, WithAsProp, withSafeTypeForAs } from '../../types'
+import { Accessibility } from '../../lib/accessibility/types'
+import { defaultBehavior } from '../../lib/accessibility'
+
+import Icon from '../Icon/Icon'
+
+export interface ToolbarItemProps
+  extends UIComponentProps,
+    ChildrenComponentProps,
+    ContentComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   */
+  accessibility?: Accessibility
+
+  /** A toolbar item can be active. */
+  active?: boolean
+
+  /** A toolbar item can show it is currently unable to be interacted with. */
+  disabled?: boolean
+
+  /** Name or shorthand for Toolbar Item Icon */
+  icon?: ShorthandValue
+
+  /**
+   * Called on click.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onClick?: ComponentEventHandler<ToolbarItemProps>
+
+  /**
+   * Called after user's focus.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onFocus?: ComponentEventHandler<ToolbarItemProps>
+
+  /**
+   * Called after item blur.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onBlur?: ComponentEventHandler<ToolbarItemProps>
+}
+
+export interface ToolbarItemState {
+  isFromKeyboard: boolean
+}
+
+class ToolbarItem extends UIComponent<WithAsProp<ToolbarItemProps>, ToolbarItemState> {
+  static displayName = 'ToolbarItem'
+
+  static className = 'ui-toolbar__item'
+
+  static create: Function
+
+  static propTypes = {
+    ...commonPropTypes.createCommon(),
+    active: PropTypes.bool,
+    disabled: PropTypes.bool,
+    icon: customPropTypes.itemShorthand,
+    onClick: PropTypes.func,
+    onFocus: PropTypes.func,
+    onBlur: PropTypes.func,
+  }
+
+  static defaultProps = {
+    as: 'button',
+    accessibility: defaultBehavior as Accessibility,
+  }
+
+  renderComponent({ ElementType, classes, unhandledProps, accessibility }) {
+    const { icon, children } = this.props
+    return (
+      <ElementType
```

Comment:
Intentionally NOT setting `disabled` attribute - according to WAI ARIA disabled items should be focusable to make screen readers happy.

Reply:
Setting `disabled` for now.